File: jagraj/22/happyNumber.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        i = 0
        while (i := i + 1) < 100:
            s = 0
            while n != 0:
                s += (n % 10) ** 2
                n = n // 10

            if s == 1:
                return True
            if s in seen:
                return False
            seen.add(s)
            n = s
        return False

File: jagraj/22/HappyNumber.java
class HappyNumber {
	private int sumDigits(int n) {
		int s = 0;
		while (n != 0) {
			s += (n % 10) * (n % 10);
			n /= 10;
		}
		return s;
	}

	public boolean isHappy(int n) {
		short[] seen = new short[101];

		int s = 0;
		while (true) {
			if (n < 100) {
				seen[s] = 1;
			}
			s = this.sumDigits(n);

			n = s;
			if (n < 100 && seen[n] == 1 || n == 1) {
				break;
			}

		}
		return n == 1;
	}
}
